core/src/main/java/net/opentsdb/data/types/numeric/aggregators/ArrayAverageFactory.java

```diff
@@ -107,13 +107,36 @@ public void accumulate(final long[] values,
       }
     }
 
+    @Override
+    public void accumulate(double value, int idx) {
+      if (Double.isNaN(value)) {
```

Comment:
We'll need the long version of the interface for the HBase stage but for this, do a `if (double_accumulator == null) <then allocate and copy from long_accumulator>` check in each `accumulate()` implementation.

Reply:
The long version of this api should be enough I guess

Reply:
I think initializing the array would be a prerequisite for this api.

Reply:
rgr